app/api/users/route.ts

```diff
@@ -81,18 +73,19 @@ export async function POST(request: NextRequest) {
         },
       ]);
 
-      if (error) {
-        throw new Error(error.message);
-      }
-
-      return new Response("Username added successfully", { status: 200 });
-    } catch (error) {
-      console.error("Error fetching user data:", error);
-      return new Response("Internal Server Error", { status: 500 });
+    if (error) {
+      throw new Error(error.message);
     }
+
+    return new Response("Username added successfully", { status: 200 });
   } catch (error) {
-    console.error("Error updating username:", error);
-    return new Response("Internal Server Error", { status: 500 });
+    console.error("Error:", error);
+
+    if (error instanceof Error) {
+      return new Response(error.message || "Internal Server Error", { status: 500 });
+    }
+
+    return new Response("Unexpected Error", { status: 500 });
```

Comment:
Why are both errors having same code?

Reply:
Both of these issues result in an Internal Server Error, which is indicated by the 500 status code.